web/src/stream_settings_components.js

```diff
@@ -24,6 +25,9 @@ export function set_right_panel_title(sub) {
     if (settings_data.using_dark_theme()) {
         title_icon_color = "#dddeee";
     }
+
+    sub.preview_url = hash_util.by_stream_url(sub.stream_id);
```

Comment:
We should pass this as a separate parameter, like `title_icon_color`, since otherwise we're mutating the object permanently.